packages/orchestrator/internal/template/build/layer/layer_executor.go

```diff
@@ -258,7 +259,7 @@ func (lb *LayerExecutor) PauseAndUpload(
 
 	// Add snapshot to template cache so it can be used immediately
 	err = lb.templateCache.AddSnapshot(
-		ctx,
+		context.WithoutCancel(ctx),
```

Comment:
Should we simply add `ctx = context.WithoutCancel(ctx)` to the top of this function? It seems like we wouldn't want anything in here to be aborted. Alternately, it's only called in `LayerExecutor.BuildLayer`, so maybe it would be cleaner to put this there:

```
	err = lb.PauseAndUpload(
		context.WithoutCancel(ctx),
		userLogger,
		sbx,
		cmd.Hash,
		meta,
	)
```